### PR TITLE
Accordion for Q&A

### PR DESCRIPTION
**Pull request by griseldasanchez**
This setups up the accordion effect for both questions and answers.

- The main items being tracked are the array of questions/answers and their lengths.
- From there, we make two more variables that will help display the ones required on the screen, figured this way we make one call to the API vs needing more make additional calls to load more. The initial length is set at the beginning, and the onClick handler will load two more at a time.  

Also sorted the questions/answers by helpfulness, and made some changes in CSS naming to prefix 'qa' where they were missing.

### CODE REVIEW

File: client/src/components/QA/QuestionsAndAnswers.jsx [hunk 1]
```diff
@@ -33,9 +45,12 @@ const QuestionsAndAnswers = () => {
         <button className="qa-search-btn">Search</button>
       </form>
 
-      <div data-testid="qa"><QACardQuestions questions={questions} /></div>
+      <div data-testid="qa"><QACardQuestions questions={questionsToShow} /></div>
 
-      <button className="qa-more">MORE ANSWERED QUESTIONS</button>
+      <button className="qa-more" onClick={() => {
```
Comment (NylonEric) [inline client/src/components/QA/QuestionsAndAnswers.jsx:50]: This is a really clean and simple way to slice the array for viewing the list. I'll implement this on my relevant components as well.

File: client/src/components/QA/QuestionsAndAnswers.jsx [hunk 2]
```diff
@@ -14,17 +14,29 @@ const QuestionsAndAnswers = () => {
...
       .then(response => {
         setQuestions(response.data);
+        setQuestionLength(response.data.length);
+        setQuestionsToShow(response.data.slice(0, questionsToShowLength));
+        setQuestionsToShowLength(questionsToShowLength + 2);
       })
       .catch(err => err);
   }, []);
 
+  // sorts questions by helpfulness
```
Comment (NylonEric) [inline client/src/components/QA/QuestionsAndAnswers.jsx:33]: Good solution for sorting.

File: client/__tests__/QA.test.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import Modal from 'react-modal';
 import { render, screen, cleanup } from '@testing-library/react';
-import QuestionsAndAnswers from '../src/components/QuestionsAndAnswers';
```
Comment (NylonEric) [inline client/__tests__/QA.test.js:4]: I notice that some of our test files use the explicit .jsx at the end of their path. I'm not sure if jest accepts both, but we may want to do one or the other for consistency.

File: client/src/components/QA/QACardAnswers.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+const QACardAnswers = ({ currentAnswers }) => {
+
+  // array for all answers for current question
+  const [answers, setAnswers] = useState([]);
+  const [answersLength, setAnswersLength] = useState(0);
+  // array for answers to render
+  const [answersToShow, setAnswersToShow] = useState([]);
+  const [answersToShowLength, setAnswersToShowLength] = useState(2);
+  // toggle is to make sure we avoid infinite rerender loop
```
Comment (NylonEric) [inline client/src/components/QA/QACardAnswers.jsx:19]: Thank you for these comments. They clarify the purpose without being excessive.

File: client/src/components/QA/QACardAnswers.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  const [toggle, setToggle] = useState(false);
+
+  useEffect(() => {
+    if (answers) {
+      for (var key in currentAnswers) {
+        answers.push(currentAnswers[key]);
+      }
+      // sets answers array
```
Comment (NylonEric) [inline client/src/components/QA/QACardAnswers.jsx:27]: This level of comment is probably closer to pseudocode and might not be necessary for those that understand the React hooks pattern.

File: client/src/components/QA/QACardAnswers.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        return b.helpfulness - a.helpfulness;
+      });
+      setAnswersLength(answers.length);
+      setAnswersToShow(answers.slice(0, answersToShowLength));
+      setAnswersToShowLength(answersToShowLength + 2);
+    }
+    setToggle(true);
+  }, []);
+
+
```
Comment (NylonEric) [inline client/src/components/QA/QACardAnswers.jsx:40]: I would leave one space here.

File: client/src/components/QA/QACardAnswers.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+
```
Comment (NylonEric) [inline client/src/components/QA/QACardAnswers.jsx:42]: Remove space?

**Review by NylonEric — APPROVED** [review]
These changes all make sense:
* Moving to sub files is much cleaner for the folder structure.
* The logic around the accordion is neat and clever, good job.
* I liked your solution for sorting and slicing the arrays.

I made a couple of comments about extra spaces and comments. Feel free to consider, but they aren't make/break changes.

File: client/__tests__/QA.test.js (hunk 3 above)
Comment (griseldasanchez) [reply client/__tests__/QA.test.js:4]: I'll go back and add '.jsx' on mine.

File: client/src/components/QA/QACardAnswers.jsx (hunk 7 above)
Comment (griseldasanchez) [reply client/src/components/QA/QACardAnswers.jsx:42]: Fixing spaces as well, thanks.

**Pull request merged**